utils/docker/Dockerfile.centos.7

```diff
@@ -140,14 +140,9 @@ RUN dnf -y install deltarpm && \
         patchelf \
         pciutils \
         python3-pip \
-        python36-Cython \
         python36-devel \
         python36-distro \
         python36-junit_xml \
-        python36-numpy \
-        python36-paramiko \
-        python36-pylint \
-        python36-requests \
```

Comment:
In general, how are we verifying that these are in fact NOT being used in any functional tests? I.e., are they only able to be used in the form of `import <package>`, etc., and so we can just look for references to the packages?

Reply:
We're not.

Docker is used for building DAOS and in some cases running NLT, but it's never been possible to run the functional tests in docker.  You can build in docker and run the functional tests using the built code, but this doesn't require the docker images to have the packages needed for running.

The centos 7 dockerfile used to be used for running pylint so it did have some of the packages (but not all), although we split out that into it's own dockerfile some time ago and could probably have removed some of these then.  I think the centos 8 dockerfile was largely a copy of the 7 version so carried forward a lot of unnecessary packages.

What we do need to be careful of however is that we don't remove things like the mpi devel packages for building optional daos components (and currently these are not build in the Fedora docker builds), there's no automated checking for this and last time I checked it the --require-optional flag needed work before we could use that to enforce it.